src/runtime/components/nuxt-img.vue

```diff
@@ -56,9 +56,23 @@ export default defineComponent({
           height: parseSize(this.height)
         }
       })
+    },
+    image (): string {
+      return this.placeholder ? this.placeholder : this.nSrc
     }
   },
-  created () {
+  mounted () {
+    if (this.placeholder) {
+      this.image = this.placeholder
```

Comment:
This is a nice start but also means on initial hydration, actual images only start loading when the whole JS bundle is loaded and hydration happens. We can enhance it later with a head script doing the same and hydrating from a `data-` attribute.

Reply:
I suppose hydrating from a `data` atrribute would make a great addition for lazy loading images.